api/src/routes/filters.rs

```diff
@@ -9,19 +10,48 @@ use std::vec;
 use utoipa::ToSchema;
 use uuid::Uuid;
 
-#[derive(serde::Deserialize, Clone, ToSchema)]
-pub struct FilterSubmissionBody {
-    pub pipeline: Option<Vec<serde_json::Value>>,
-    pub permissions: Option<Vec<i32>>,
-    pub catalog: Option<String>,
+#[derive(serde::Deserialize, serde::Serialize, Clone, ToSchema)]
+pub struct FilterVersion {
+    fid: String,
+    pipeline: String,
 }
 
 #[derive(serde::Deserialize, serde::Serialize, Clone, ToSchema)]
-struct Filter {
-    pub pipeline: Vec<serde_json::Value>,
+pub struct Filter {
+    #[serde(rename = "_id")]
+    pub id: String,
     pub permissions: Vec<i32>,
+    pub user_id: String,
     pub catalog: String,
+    pub active: bool,
+    pub active_fid: String,
+    pub fv: Vec<FilterVersion>,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Clone, ToSchema)]
+pub struct FilterPublic {
+    #[serde(rename(serialize = "id", deserialize = "_id"))]
     pub id: String,
+    pub permissions: Vec<i32>,
+    pub user_id: String,
+    pub catalog: String,
+    pub active: bool,
+    pub active_fid: String,
+    pub fv: Vec<FilterVersion>,
+}
+
+impl FilterPublic {
+    pub fn from_filter(filter: Filter) -> Self {
```

Comment:
Couldn't figure out any way to avoid this, since we have to serialize a response, and we don't want the underscore there.

Reply:
Don't we `deserialize` when we send to Mongo? In which case maybe you can just have the one private filter struct.